### ISSUE TITLE
not all clients in sync

### ISSUE THREAD

**Issue by srulikuk** [meganz/MEGAcmd#46]
I setup mega-cmd in 2 locations, location 1 is where the files are created so primarily files are uploaded in that location. The other location is download only (i use rsync at that location to have daily offline backup).
However i am struggling to get them to be fully in sync. output of mega-sync in upload location is; 

`ID LOCALPATH   REMOTEPATH     ActState   SyncState       SIZE         FILES    DIRS`
 `0 /path/to/0/           /remote/0/            Active        Synced      16.03 GB    190105   4510`
 `1 /path/to/1/           /remote/1/            Active        Synced        3.08 GB        7778     397`
 `2 /path/to/2/           /remote/2/            Active        Synced        9.07 GB            74       40`

mega-sync in location 2 shows;
`ID LOCALPATH      REMOTEPATH     ActState   SyncState     SIZE  FILES   DIRS`
 `0 /path/to/0/ /remote/0/   Active     Synced    14.63 GB 174104   4510`
 `1 /path/to/1/ /remote/1/   Active     Synced     3.08 GB   7778    397` 
 `2 /path/to/2/ /remote/2/   Active     Synced     9.07 GB     74     40`

in location 2 mega-transfers --only-downloads --show-syncs returns empty so nothing is being downloaded.
The problem primarily seems to be with ID 0 as the others seem to be in sync, how do i get to the root of this?

**Comment by polmr** [comment]
@srulikuk , that's odd ... do you have the latest version of MEGAcmd in both locations? 
Is the MEGA account in both cases is the same? 
- If affirmative, if you login in the webclient, and you browse into `/remote/` and right-click->Info on "0", what does it say?
- If negative, are you missing any particular file? user in location 2 has permissions to all paths in `/path/to/0` ?

**Comment by srulikuk** [comment]
I guess I have the latest, I downloaded from mega.nz this week and installed them. (not sure where to check version).
I use the same account in both locations.
On the web client the info shows the full amount as shown in the upload location 
`16.03 GB 190105 4510`

I just noticed another issue, not sure if its related, sync ID 2 do show there are in sync but mega shows `InitScan Synced 9.07 GB 74 40`
however;
`du -hcs /path/to/2/`
`9.6G`
`find /path/to/2/ -type f | wc -l`
`78`
it has been at initscan for hours now

**Comment by srulikuk** [comment]
in location 2 in the terminal that i started mega-cmd i now see 1000's if lines
`[mp3 @ 0x7f3377d07920] Header missing`
Every few thousand lines followed by
'[mp3 @ 0x7f3377d05d70] decoding for stream 0 failed`
`[mp3 @ 0x7f3377d05d70] Could not find codec parameters for stream 0 (Audio: mp3, 0 channels, s16p): unspecified frame size`
`Consider increasing the value for the 'analyzeduration' and 'probesize' options`

not sure what that means

**Comment by srulikuk** [comment]
With regards to the original problem of sync ID 0, i have some more info which makes it even more difficult for me to understand, mega-sync in location 1 and info on web client show the same, however in location 1 (where files originate) 
`du -hcs /path/to/0` 16G
`find /path/to/0 -type f | wc -l` 174104
this same as in location 2 (download only), so in location 1 and on webclient mega-sync seems to be showing more files than there should be but correct size, location 2 shows correct number of files but smaller size.

In Summary 3 issues, not sure if they are related.
1. sync ID 2 for some reason stopped syncing (still on initscan)
2. sync ID 0 in upload location showing much more files than exists but correct size and in download location shows correct number of files and wrong size.
3. The "mp3 @" error i posted above

**Comment by srulikuk** [comment]
@polmr 
I guess I have the latest version, I downloaded from mega.nz this week and installed them. (not sure where to check version).
I use the same account in both locations.

I have multiple issues, not sure if they are related, 
sync ID 2 shows
`InitScan Synced 9.07 GB 74 40`
its still at InitScan after 36 hours and is still showing old file count, not uploading new files.

in location 2 in the terminal that i started mega-cmd i now see 1000's of lines
`[mp3 @ 0x7f3377d07920] Header missing`
`Every few thousand lines followed by`
`[mp3 @ 0x7f3377d05d70] decoding for stream 0 failed`
`[mp3 @ 0x7f3377d05d70] Could not find codec parameters for stream 0 (Audio: mp3, 0 channels, s16p): unspecified frame size`
`Consider increasing the value for the 'analyzeduration' and 'probesize' options`
not sure what that means

With regards to the original problem of sync ID 0, i have some more info which makes it even more difficult for me to understand, mega-sync in location 1 and info on web client show the same, however in location 1 (where files originate)
location 1 (upload)
`du -hcs /path/to/0 16G`
`find /path/to/0 -type f | wc -l 174104`
location 2 (download)
`du -hcs /path/to/0 16G`
`find /path/to/0 -type f | wc -l 174104`
so "location 2" has all the files of sync 0 but "location 1" mega-sync and on webclient seem to be showing more files real count but correct size, location 2 mega-sync shows correct "real" number of files but smaller total size.

In Summary 3 issues, not sure if they are related.

sync ID 2 for some reason stopped syncing (still on initscan after more than 36 hours)
sync ID 0 in upload location showing much more files than exists but correct size and in download location shows correct number of files and wrong size.
The "mp3 @" error

**Comment by polmr** [comment]
Hi @srulikuk , apologies for the delay...
To see the version you can type `version` in MEGAcmd shell.
Regarding the issue with the "mp3 @" errors ... We use ffmpeg to create thumbnails of videos. It seems to be having trouble with some particular file and prints warning messages in the console. I have only seen that in mac. Is that your OS? we will try to figure out a way to avoid that.
Regarding the InitScan issue. It seems that something is holding the scan of files. We'd need to see some logs in order to figure out what could that be (permission issues, a locked file, ...)
Regarding the issue with the number of files: some files are skipped, `exclude` gives you the list. In any case, you seem to be having more files in the cloud that locally, not the contrary. Certain operations could have created duplicated files in the cloud. That could explain the discrepancy between the 174104 and 190105. However, if /remote/0 is the same in Location 1 & Location 2. The number in both cases should be the same, unless for some reason the cache is broken. Please, use `reload` command and see if the issue persist. If so, please tell us.
Please, tell us what is your OS in both locations and we will give you instructions to produce logs.

**Comment by srulikuk** [comment]
thanks for your response @polmr.
version - MEGA CMD version: 0.9.9: code 90900 (i thought it will be in the shell but i try to avoid going in as EXIT really is EXIT not as the help pages suggest)
OS - Ubuntu 16.04 in both locations, (location 1 (upload) is server no desktop env)
mp3 error - here is the cause of it... It's possible that there is 1 or more movie files, however all the files are encrypted with encFS (content only not file names) so that explains ffmpeg's problem, is there anyway to turn the ffmpeg function off?
excluded - there are no excluded files in path/0
filecount - the file system in both locations report same number of files, only mega-sync in location 1 and online show more, if there are duplicated files in the cloud why is it only showing in location 1?
InitScan - after a reboot it seemed to update, however mega-sync then stops showing updated file count and size but it does sync (i.e. 11.52G 100files, real number is 14G 140files) in location 2 i see the new files, i am not sure if this is only with path/2 and the other 2 syncs have too many files for me to be able to verify if it is uploading new files.

**Comment by polmr** [comment]
>  is there anyway to turn the ffmpeg function off?
I'm afraid currently there is no way to do that. I don't think the encFS is the problem here, it could be an mp3 which ffmpeg simply can't decode for whatever the reason. 
> why is it only showing in location 1?
That's the cuestion. It is the same code doing the counting and using the same information. It shouldn't. You did try the `reload` command. Didn't you?
Here are the instructions for creating a log file in linux so that we can try and analyse what's going on:
Please, close all megacmd instances and execute the server in a separate console:
Open a console and type
```
mega-quit
mega-cmd-server --debug-full | tee megacmd.log
```
Then, open another console and execute mega-cmd interactive shell:
```
mega-cmd
```
Then leave it open for a while (~10 minutes). 
Quit mega-cmd-server then (Ctrl+C or typing mega-quit in a third console)

Then please send us that megacmd.log file. The log might contain some sensitive data. Feel free to remove anything you don't want to share with us.
If you do not want to post it here (you'd rather not), send it to support@mega.nz and refer this issue in your email so that it can reach development team.

**Comment by srulikuk** [comment]
>  it could be an mp3 which ffmpeg simply can't decode for whatever the reason.
>
There is no way ffmpeg would be able to decode an mp3 that encrypted with encFS, that is the issue, now that we know the issue, i am not really worried about it.

> You did try the reload command. Didn't you?
>
No but i rebooted the server and also used the mega-exclude with --restart-syncs flag.

> Then leave it open for a while (~10 minutes).
>
I Will do it in the next day or so.

**Comment by srulikuk** [comment]
@polmr 
I have emailed you a link to the log files (1 for each location) they are about 48MB each, before step 2 (run mega-cmd in another console) i let the first command finish scanning all the files, that took about 45 minutes (the files are on a old slow nas device hence the slow progress).

There was no file activity at the time when i ran it so when it finished initialising even so i ran mega-cmd there was no real output to the log file.

**Comment by srulikuk** [comment]
@polmr did you check the log files? did you find anything to explain the situation?

**Comment by polmr** [comment]
Hi @srulikuk , apologies for the late response. I've been analyzing the logs, but I'm afraid that I haven't found anything suspicious. May I suggest you to list the files in `/remote/0/` and have them counted in both location 1 & 2, executing this in a console:
```
mega-reload
mega-find /remote/0 | sort | wc -l
```
You can also save the output to a file and `diff` them, to see what are the missing files. The mega-reload should fix any issues in case the cache has been broken anyhow. 
```
mega-find /remote/0 | sort > list.txt
```
Regarding the `mp3` errors, we have added some code to prevent such lines from being printed. It will be available in the next release.

**Comment by srulikuk** [comment]
@polmr 
mega-find seems very unreliable, in location 1 "mega-find /remote/0 | sort | wc -l" mostly gives the same result, however in location 2 it's all over the place, see attached output of location 2
[mega-log.txt](https://github.com/meganz/MEGAcmd/files/2066181/mega-log.txt)

**Comment by polmr** [comment]
Hi @srulikuk , I'm puzzled here ... `mostly gives the same result`. If no files are added/deleted, it should give the same result always!. The results for location 2 are as odd as it gets. We should focus on this particular issue and try to get the output of mega-find whenever the count of files+folders is != 8121 in /remote1. 
Can you launch the server manually to get logs (from a console):
```
killall mega-cmd-server
mega-cmd-server --debug-full | tee megacmd.log
```
And then do (in another console)
```
mega-find /remote1 > remote1files.txt || echo "mega-find unexpected exit: $?"; 
cat remote1files.txt | sort | wc -l
```
until the count is different than the expected, and give send us both megacmd.log & remote1files.txt. Again, you'd rather send these files to support@mega.nz refering to this issue.

**Comment by polmr** [comment]
Thanks @srulikuk , We will analyze the logs as soon as they arrive to the development team.
```
Also when starting mega-cmd-server why does it "Creating thumb/preview" for all files? is there anyway to disable this? and why is it "Restoring missing attributes" for all files? these 2 take about 20 minutes for mega to finish startup.
```
Upon startup, the sync engine checks for images that have no thumbnail and attempts to create them in their absence. If the creation of the thumbnail fails (this could happen for many reasons: incorrect format, malformed image, issue with permissions, ...). We take the conservative approach to try again next time. In any case, it seems as if you had too many files without thumbnail. If you see any particular file that always is attempted and you don't mind you can share it with us. In any case, we are currently working on some changes in the `sdk` to do all that in the background and avoid the waiting. It will probably be live in the next release. 

Your guess concerning files starting with `.` is correct: they are excluded by default. 
```
So it seems the all is syncing correctly the only problem is the accuracy of the mega reporting tools, which must be resolved
```
You're right, mega-find output should be correct and the count of files via "mega-sync" should be correct. Let's hope we can figure out how to reproduce your issue (we have never had any issues with it before).

**Comment by srulikuk** [comment]
`We will analyze the logs as soon as they arrive to the development team`
it's ticket reference #FNB-350-83230

**Comment by srulikuk** [comment]
`If you see any particular file that always is attempted and you don't mind you can share it with us.`
All the files, I mentioned this last time with the MP3 issue, all the files are encrypted, there is no way you can get thumbnail or anything else from it (the file names are not encrypted only the content)
It would be great if you can add an option to disable thumbnails, etc

**Comment by polmr** [comment]
^ I see that. Currently there's no way to disable thumbnails and previews. That's obviously interesting in your case. We will add the option asap.

**Comment by srulikuk** [comment]
`Regarding the mp3 errors, we have added some code to prevent such lines from being printed. It will be available in the next release.`
`We will add the option asap.`
How do I know when new versions are released? how do I update? will a regular update/dist-upgrade update mega-cmd-server?

**Comment by srulikuk** [comment]
^ great

**Comment by polmr** [comment]
The logs didn't indicate anything useful. However I've replicated your file structure with empty files and I occasionally can reproduce it. Seems an issue with the outputting, we will try to fix it for the next update. Thanks for reporting @srulikuk.

**Comment by srulikuk** [comment]
@polmr thanks, please advise when update with fix is released

**Comment by polmr** [comment]
Should be fixed already
